Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -321,18 +358,40 @@ public async Task<string> WalkTree(string treeNodeKey)
                     }
                     finally
                     {
-                        // Always call this callback, whether or not visit node was successful.
-                        await this.Parameters.Callbacks.AfterVisitNode(
+                        if (this.Parameters.CallbacksV2 != null)
+                        {
+                            // When CallbacksV2 is not null, continues with the new ITreeWalkerCallbacksV2.
+                            treeNodeContext = new TreeNodeContext(
+                                this.Parameters.SessionId,
+                                current,
+                                await this.EvaluateDynamicProperty(this.Schema.Tree[current].Properties, null),
+                                this.Parameters.UserContext,
+                                this.walkTreeCts.Token,
+                                this.Parameters.TreeName,
+                                this.Parameters.RootSessionId,
+                                this.currentNodeSkipActionContext
+                            );
+
+                            await this.Parameters.CallbacksV2.AfterVisitNode(treeNodeContext).ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            // Always call this callback, whether or not visit node was successful.
```

Comment:
>// Always call this callback, whether or not visit node was successful. [](start = 28, length = 71)

Let's move this comment so it is valid for V2 or V1. Perhaps above the if statement.

// Always call AfterVisitNode, even if VisitNode threw exception.

Reply:
make sense. fixed

---
In reply to: [505918812](https://github.com/microsoft/Forge/pull/32#discussion_r505918812) [](ancestors = 505918812)